oak-core/src/main/java/org/apache/jackrabbit/oak/plugins/index/cursor/TraversingCursor.java

```diff
@@ -159,10 +159,15 @@ private void fetchNext() {
 
                 readCount++;
                 if (readCount % 1000 == 0) {
-                    FilterIterators.checkReadLimit(readCount, settings);
-                    String caller = IndexUtils.getCaller(this.settings.getIgnoredClassNamesInCallTrace());
-                    LOG.warn("Traversed {} nodes with filter {} called by {}; consider creating an index or changing the query" , 
-                            readCount, filter, caller);
+                    if (readCount == 20000) {
+                        LOG.warn("Traversed {} nodes with filter {}; consider creating an index or changing the query",
+                                readCount, filter, new Exception("call stack"));
```

Comment:
Why do you log the full stack trace? Isn't logging the caller (as it is done a few lines alter by 
```
String caller = IndexUtils.getCaller(this.settings.getIgnoredClassNamesInCallTrace());
```

good enough? The stack trace creates a lot more noise, and I don't see it to be more actionable in the first place.

Reply:
Not it is not good enough.

> The stack trace creates a lot more noise

But it is needed. In the past, we have seem multiple cases where the caller was not sufficient. The stack trace is needed.

Reply:
Well, but then we have a inconsistent reporting here, and we should harmonize it. And then this magic number ``20000`` makes me wonder what is so special about it (and why this exception is not reported at 10k or 80k)?

What information is needed for analysis? I think that the full stack trace is always noise, but you just need more information than currently provided by ``IndexUtils.getCaller()``. What information is this? Mabye 5 more lines of the stack below that line which is currently reported?